packages/jellyfish-api-core/__tests__/category/account/futureswap.test.ts

```diff
@@ -0,0 +1,744 @@
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_current']).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_burned']).toBeUndefined()
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_minted']).toBeUndefined()
+
+      // dfip2203 burn should be empty
+      const burnBefore = await testing.rpc.account.getBurnInfo()
+      expect(burnBefore.dfip2203).toStrictEqual([])
+
+      {
+        // check contractAddress
+        const balance = await testing.rpc.account.getAccount(contractAddress)
+        expect(balance).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+      }
+
+      {
+        // check tslaAddress
+        const balance = await testing.rpc.account.getAccount(tslaAddress)
+        expect(balance).toStrictEqual([])
+      }
+    }
+
+    // get minted DUSD
+    const dusdMintedBefore = (await testing.rpc.token.getToken(idDUSD))[idDUSD].minted
+
+    // move to next settle block
+    const nextSettleBlock = await getNextSettleBlock()
+    await testing.generate(nextSettleBlock - await testing.rpc.blockchain.getBlockCount())
+
+    let mintedDUSD: BigNumber
+    // check future settled
+    {
+      // calclulate minted DUSD. dtoken goes for a discount.
+      mintedDUSD = new BigNumber((1 - futRewardPercentage) * 2 * swapAmount).dp(8, BigNumber.ROUND_FLOOR) // (1 - reward percentage) * TSLADUSD value * TSLA swap amount;
+      const dusdMintedAfter = (await testing.rpc.token.getToken(idDUSD))[idDUSD].minted
+      expect(dusdMintedAfter).toStrictEqual(dusdMintedBefore.plus(mintedDUSD))
+
+      const pendingFutures = await testing.container.call('listpendingfutureswaps')
+      expect(pendingFutures.length).toStrictEqual(0)
+
+      // check live/economy/dfip2203_*
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_current']).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_burned']).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_minted']).toStrictEqual([mintedDUSD.toFixed(8) + '@DUSD'])
+
+      {
+        // check contractAddress
+        const balance = await testing.rpc.account.getAccount(contractAddress)
+        expect(balance).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+      }
+
+      {
+        // check tslaAddress
+        const balance = await testing.rpc.account.getAccount(tslaAddress)
+        expect(balance).toStrictEqual([mintedDUSD.toFixed(8) + '@DUSD'])
+      }
+    }
+
+    // check burn
+    const burnAfter = await testing.rpc.account.getBurnInfo()
+    expect(burnAfter.dfip2203).toStrictEqual([swapAmount.toFixed(8) + '@TSLA'])
+
+    // check results can be retrived via account history
+    const accountHistories = await testing.rpc.account.listAccountHistory('all', { txtype: DfTxType.FUTURE_SWAP_EXECUTION })
```

Comment:
@surangap this contract execution will have txid?

Reply:
no real txid since execution and refund happens at scheduled future time from creation. But when you query using `listAccountHistory` it will be as follow 
```
[{"owner":"bcrt1qpcmmnjr5urgnymakmjzx4733k9c0fjyp3athht","blockHeight":125,"blockHash":"a90df4c605c3ed3cfa07099eb1e1f2de6d863211885c28ffdcf8a49f45599ac1","blockTime":1649247386,"type":"FutureSwapExecution","txn":4294967295,"txid":"0000000000000000000000000000000000000000000000000000000000000000","amounts":["1.90000000@DUSD"]},{"owner":"bcrt1qpcmmnjr5urgnymakmjzx4733k9c0fjyp3athht","blockHeight":125,"blockHash":"a90df4c605c3ed3cfa07099eb1e1f2de6d863211885c28ffdcf8a49f45599ac1","blockTime":1649247386,"type":"FutureSwapExecution","txn":4294967294,"txid":"0000000000000000000000000000000000000000000000000000000000000000","amounts":["1.90000000@DUSD"]}]
```